src/core/xprops.c

```diff
@@ -894,8 +899,9 @@ size_hints_from_results (GetPropertyResults *results,
   hints->min_height = cvtINT32toInt (raw->minHeight);
   hints->max_width  = cvtINT32toInt (raw->maxWidth);
   hints->max_height = cvtINT32toInt (raw->maxHeight);
-  hints->width_inc  = cvtINT32toInt (raw->widthInc);
-  hints->height_inc = cvtINT32toInt (raw->heightInc);
+  /* Convert the increments to application pixels */
+  hints->width_inc  = cvtINT32toInt (raw->widthInc) / scale;
+  hints->height_inc = cvtINT32toInt (raw->heightInc) / scale;
```

Comment:
Why only `width_inc` and `height_inc`? If these are scaled then I would expect that other things are scaled too... It just seems weird to keep mixed things in one struct, either you unscale all or nothing.